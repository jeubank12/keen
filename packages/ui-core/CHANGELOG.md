# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.5.0](https://github.com/keen/keen/compare/@keen.io/ui-core@0.4.0...@keen.io/ui-core@0.5.0) (2020-05-07)


### Features

* 🎸 arc animation ([267bb82](https://github.com/keen/keen/commit/267bb82b3fc56649daa95f38a88139a3b85c6e3e))





# [0.4.0](https://github.com/keen/keen/compare/@keen.io/ui-core@0.3.1...@keen.io/ui-core@0.4.0) (2020-05-04)


### Features

* 🎸 embeded registration components ([69ad26b](https://github.com/keen/keen/commit/69ad26beaa95fcd07297adc45bd1f09137034209))
* 🎸 registration redirect ([1c8363c](https://github.com/keen/keen/commit/1c8363c204829356ba2ac1924c9b0eca12d0ea2d))





## [0.3.1](https://github.com/keen/keen/compare/@keen.io/ui-core@0.3.0...@keen.io/ui-core@0.3.1) (2020-04-29)


### Bug Fixes

* 🐛 slider fix ([66b6321](https://github.com/keen/keen/commit/66b63213c2f592fcb462d18ab7a475eced32b695))





# [0.3.0](https://github.com/keen/keen/compare/@keen.io/ui-core@0.2.1...@keen.io/ui-core@0.3.0) (2020-04-29)


### Bug Fixes

* 🐛 issue with x property for slider ([43e71f0](https://github.com/keen/keen/commit/43e71f03f0496cdfb107b93859b3f851ca4aae62))
* 🐛 range slider minimum and maximum ([d1bb5dd](https://github.com/keen/keen/commit/d1bb5dd09ab94702bea330edcf60eef2eb3c29dd))


### Features

* 🎸 handle click on slider ([f696c7f](https://github.com/keen/keen/commit/f696c7f2a9d1fd1b2d06e92134b3298317446570)), closes [#269](https://github.com/keen/keen/issues/269)





## [0.2.1](https://github.com/keen/keen/compare/@keen.io/ui-core@0.2.0...@keen.io/ui-core@0.2.1) (2020-04-27)

**Note:** Version bump only for package @keen.io/ui-core





# [0.2.0](https://github.com/keen/keen/compare/@keen.io/ui-core@0.1.0...@keen.io/ui-core@0.2.0) (2020-04-15)


### Bug Fixes

* 🐛 tooltip position ([93c14de](https://github.com/keen/keen/commit/93c14dedfe5e499c13cc1bc0ca3630e4f3f65a22))


### Features

* 🎸 interval slider ([a98b284](https://github.com/keen/keen/commit/a98b28402d227783fe68813c86d389d74bd46b10))
* 🎸 slider component ([4cc7267](https://github.com/keen/keen/commit/4cc7267f0d1512437d3c6710d17078c6aa488a45)), closes [#173](https://github.com/keen/keen/issues/173)
* 🎸 table chart plot ([91d3a56](https://github.com/keen/keen/commit/91d3a5618d1f534d3b11e4f3e95596da7c5c8033)), closes [#204](https://github.com/keen/keen/issues/204)
* 🎸 toggle ([bb7f910](https://github.com/keen/keen/commit/bb7f9108c2ab50ab14a322ac36c2f816ffa5d3ae)), closes [#240](https://github.com/keen/keen/issues/240)
* 🎸 vertical tooltip position ([8345674](https://github.com/keen/keen/commit/8345674e8a7ccffc42dd8688bff9b86d95be296c)), closes [#168](https://github.com/keen/keen/issues/168)





# 0.1.0 (2020-03-16)


### Features

* 🎸 bar charts ([c8eba7a](https://github.com/keen/keen/commit/c8eba7aa3b9223d97f16a70bc62bf1daa69da689))
* 🎸 border radius for card component ([5657be2](https://github.com/keen/keen/commit/5657be2ce44bf0dcd66b5af31a777205d6c5e272)), closes [#3](https://github.com/keen/keen/issues/3)
* 🎸 cartesian grid ([09433a7](https://github.com/keen/keen/commit/09433a7c5d7a75028b3c7078844711c652b99c8f))
* 🎸 funnel chart ([4adc4e3](https://github.com/keen/keen/commit/4adc4e3cca98f8855b57173dac67cd840fe09d2d))
* 🎸 funnel chart ([78d6f37](https://github.com/keen/keen/commit/78d6f372f83565ce547a6ee6e8af9709ba7d1f9d))
* 🎸 integrate text component ([fa84aa2](https://github.com/keen/keen/commit/fa84aa2b807265bdef394404c8d6347ffb2b3c48))
* 🎸 legend ([386f140](https://github.com/keen/keen/commit/386f140b9c1626807988d152470708fed6075c56))
* 🎸 metric ([52a0769](https://github.com/keen/keen/commit/52a07699a95b190a78429bad2d0a40d838bb1358))
* 🎸 pie chart ([1ec6953](https://github.com/keen/keen/commit/1ec6953d04d5d2dd9f487f7d371c2fb9637b0eb8))
* 🎸 pie chart ([e1063a9](https://github.com/keen/keen/commit/e1063a9db7c8a477b9bcc85f693442eece0c410d))
* 🎸 responsive wrapper ([7bd729a](https://github.com/keen/keen/commit/7bd729afddb3df58e67e36c22a96a9bee497f803))
* 🎸 support for typography in tooltip ([472260e](https://github.com/keen/keen/commit/472260e04c878db1b454b68c8cfc3812d0d22ddf))
* 🎸 text component ([94d6a43](https://github.com/keen/keen/commit/94d6a432e5ff6124b052e71043de6e767a6dd009))
* 🎸 tooltip component ([2ce4a69](https://github.com/keen/keen/commit/2ce4a69ec7f981efa69f84562442f00f749a6d0c))
