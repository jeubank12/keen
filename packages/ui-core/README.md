# `@keen.io/ui-core`

[![Known Vulnerabilities](https://snyk.io/test/github/keen/keen/badge.svg?targetFile=packages/ui-core/package.json)](https://snyk.io/test/github/keen/keen?targetFile=packages/ui-core/package.json)

Set of reusable visual `React` components used in `keen.io` projects.
