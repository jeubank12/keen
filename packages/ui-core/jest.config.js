module.exports = {
  ...require('../../jest.config.base'),
  setupFilesAfterEnv: ['<rootDir>/jest.setup.ts'],
  moduleNameMapper: {
    '^@keen.io/(.*)$': '<rootDir>/../$1/src',
    '\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$':
      '<rootDir>/__mocks__/fileMock.js',
  },
};
