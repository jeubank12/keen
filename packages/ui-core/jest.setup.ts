import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import '@testing-library/jest-dom';
import './matchMedia.mock';

Enzyme.configure({ adapter: new Adapter() });
