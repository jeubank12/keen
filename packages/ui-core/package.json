{
  "name": "@keen.io/ui-core",
  "version": "0.24.2",
  "description": "Keen visual components",
  "author": "Keen.IO <team@keen.io> (https://keen.io/)",
  "homepage": "https://github.com/keen/keen#readme",
  "license": "MIT",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "module": "src/index.ts",
  "directories": {
    "lib": "dist"
  },
  "files": [
    "dist"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/keen/keen.git"
  },
  "scripts": {
    "test": "jest",
    "prebuild": "rimraf dist && rimraf tsconfig.tsbuildinfo",
    "build": "tsc"
  },
  "bugs": {
    "url": "https://github.com/keen/keen/issues"
  },
  "dependencies": {
    "@keen.io/colors": "^0.2.0",
    "@keen.io/icons": "^0.4.3",
    "color": "^3.1.2",
    "framer-motion": "^1.11.1",
    "polished": "^3.4.4",
    "react": "^16.12.0",
    "react-column-resizer": "^1.1.9",
    "react-dom": "^16.12.0",
    "react-select": "^3.1.0",
    "styled-components": "5.1.0",
    "styled-system": "5.1.5",
    "webfontloader": "^1.6.28"
  },
  "devDependencies": {
    "@testing-library/jest-dom": "^5.10.1",
    "@testing-library/react": "^10.3.0",
    "@types/color": "^3.0.1",
    "@types/react-select": "^3.0.13",
    "@types/styled-components": "5.1.0",
    "@types/styled-system": "5.1.9",
    "@types/webfontloader": "^1.6.29"
  },
  "publishConfig": {
    "access": "public"
  }
}
