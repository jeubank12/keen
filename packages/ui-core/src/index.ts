import { FontLoader, Title, Text, RefText } from './typography';

import {
  Typography,
  Position,
  Alignment,
  Layout,
  ColorMode,
  RangeType,
  CeilType,
  SortMode,
  SortByType,
} from './types';

import {
  Alert,
  Badge,
  Button,
  BulletList,
  Group,
  Card,
  CardSettings,
  Checkbox,
  ColorAdjuster,
  Error,
  Label,
  Loader,
  Input,
  Tooltip,
  TooltipMode,
  RangeSlider,
  IntervalSlider,
  calculateIntervalValue,
  ColorScale,
  Toggle,
  PasswordInput,
  PasswordValidator,
  Ruler,
  RulerTick,
  Ceil,
  HeaderCeil,
  SortArrows,
  RESIZE_ELEMENT_WIDTH,
  ARROW_SIZE,
  Sticky,
  DragLine,
  TableCell,
} from './components';

export {
  Alert,
  Badge,
  Button,
  Alignment,
  BulletList,
  ColorAdjuster,
  Card,
  CardSettings,
  Checkbox,
  Error,
  Title,
  Text,
  RefText,
  Group,
  Label,
  Loader,
  Typography,
  FontLoader,
  Tooltip,
  TooltipMode,
  Input,
  Position,
  PasswordInput,
  PasswordValidator,
  Layout,
  ColorMode,
  RangeSlider,
  IntervalSlider,
  calculateIntervalValue,
  ColorScale,
  RangeType,
  Ruler,
  RulerTick,
  Toggle,
  Ceil,
  HeaderCeil,
  SortArrows,
  CeilType,
  SortMode,
  SortByType,
  Sticky,
  DragLine,
  TableCell,
  ARROW_SIZE,
  RESIZE_ELEMENT_WIDTH,
};
