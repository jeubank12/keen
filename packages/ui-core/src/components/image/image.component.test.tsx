import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';

import { Loader } from '@keen.io/ui-core';

import Image from './image.component';

describe('@keen.io/ui-core - <Image />', () => {
  it('should render image', () => {
    const wrapper = mount(<Image name="test" />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render image from provided src', () => {
    const src = 'http://image.com';
    const wrapper = mount(<Image name="test" src={src} />);

    expect(wrapper.html()).toEqual(`<img src="${src}">`);
  });

  it('should render loader once image is not loaded yet', () => {
    const src = 'http://image.com';
    const wrapper = mount(<Image name="test" src={src} useLoader />);

    expect(wrapper.find(Loader).length).toBeTruthy();
  });
});
