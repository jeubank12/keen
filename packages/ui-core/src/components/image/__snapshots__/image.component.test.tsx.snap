// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@keen.io/ui-core - <Image /> should render image 1`] = `
<Component
  name="test"
>
  <img
    onLoad={[Function]}
    src="/test"
  />
</Component>
`;
